src/LinearOperators.jl

```diff
@@ -244,21 +260,22 @@ function transpose(op :: AbstractLinearOperator{T}) where T
     ctprod = op.ctprod
   end
 
-  return LinearOperator{T}(op.ncol, op.nrow, op.symmetric, op.hermitian,
-                           v -> conj(ctprod(conj(v))), # A.'v = conj(A' conj(v))
-                           op.prod,                    # (A.').' = A
-                           w -> conj(op.prod(w)))      # (A.')' = conj(A)
+  prod = @closure v -> conj.(ctprod(conj.(v)))  # A.'v = conj(A' conj(v))
+  ctprod = @closure w -> conj.(op.prod(w))       # (A.')' = conj(A)
+  F4 = typeof(prod)
+  F5 = typeof(ctprod)
+  return LinearOperator{T,F4,F1,F5}(op.ncol, op.nrow, op.symmetric, op.hermitian, prod, op.prod, ctprod)
```

Comment:
Using `newprod = @closure` and `newctprod = @closure` on lines 263 and 264 and updating lines 265-267 accordingly, the transpose of Hermitian test passes.
That means that `@closure` is mixing `ctprod`s, possibly?